### PR TITLE
Replace flake8 and move isort on Python files to ruff

### PR DESCRIPTION
**Pull request by flpm**
Issue number of the reported bug or feature request: #45

**Describe your changes**
- Removed pre-commit check for flake8
- Removed setup.cfg (flake8 config)
- Added a pre-commit check for `ruff` that cover flake8 checks and isort for Python files
- Prevented isort pre-commit check from running on Python files

**Testing performed**
- Verified that ruff is fixing import order in .py and .pyi files (tested by introducing wrong order)
- Verified that ruff is fixing flake8 rules (tested by introducing unused import)

**Additional context**
Some `isort` options are not supported by `ruff`.

Option `multi_line_output = 3` (break multiple imports in multiple lines using parenthesis) is not supported by ruff. The default behavior is the same as `multi_line_output = 7` (all imports in one line without breaks), which used along `force-single-line = true` causes the multiple import to be split into individual imports. For example:
```
from collections import Counter, defaultdict, deque
```
is reformatted to:
```
from collections import Counter
from collections import defaultdict
from collections import deque
```
Options `include_trailing_comma = true` and `force_grid_wrap = 0` are related to `multi_line_output = 3` and are not applicable in ruff.

Option `use_parentheses = true` is also not supported by ruff, but because imports are in individual lines and `line-length = 95`, removing it would be very unlikely to cause problems.

Some references:
[Use Parentheses in isort documnentation](https://pycqa.github.io/isort/docs/configuration/options.html#use-parentheses)
[Multi Line Output Modes in isort documentation](https://pycqa.github.io/isort/docs/configuration/multi_line_output_modes.html)
[Flake 8 rules](https://www.flake8rules.com/)
[Ruff rules](https://beta.ruff.rs/docs/rules/)

### CODE REVIEW

**Comment by godlygeek** [comment]
@flpm Thank you! Would you mind editing your commit messages to add the [DCO signoff](https://github.com/bloomberg/pystack/blob/main/CONTRIBUTING.md#contribution-licensing)?

From the state your PR is in now, you can run:
```shell
git commit --amend -s
git push -f
```
to add a DCO signoff added to your most recent commit and then force-push it back to the PR.

**Comment by flpm** [comment]
After PR #77 is merged I will rebase and fix the code, I will make sure all the commits have the DCO signoff :+1:

**Comment by godlygeek** [comment]
I've merged #77 so you should now be unblocked, @flpm!

File: pyproject.toml
Comment (gusmonod) [inline pyproject.toml:23]: Do we feel strongly about keeping the import style the same as it is? I personally don't mind either way, as long as we have a consistent and enforceable style in the long run. Maybe we could remove `isort` altogether and use `ruff`'s convention instead?

It looks like these three options were introduced by Pablo when the project was started and never changed since then.

Comment (flpm) [reply pyproject.toml:23]: That’s right. The Makefile (before the cleanup) had flake8 running on Python files and isort running on both Python and Cython files.

**Review by godlygeek — APPROVED** [review]
LGTM

**Comment by godlygeek** [comment]
Thanks for seeing this through, @flpm!

**Pull request merged**
